contracts/margined_engine/src/reply.rs

```diff
@@ -34,20 +31,26 @@ pub fn update_position_reply(
     env: Env,
     input: Uint128,
     output: Uint128,
+    position_id: u64,
     reply_id: u64,
 ) -> StdResult<Response> {
-    let mut swap = read_tmp_swap(deps.storage)?;
-
-    let position_key = keccak_256(&[swap.vamm.as_bytes(), swap.trader.as_bytes()].concat());
-
-    let mut position = get_position(
-        deps.storage,
-        &position_key,
-        &swap.vamm,
-        &swap.trader,
-        &swap.side,
-        env.block.height,
-    )?;
+    let mut swap = read_tmp_swap(deps.storage, &position_id.to_be_bytes())?;
+
+    let mut position: Position = Position {
```

Comment:
should update `fn get_position`

Reply:
fn **get_position** is not necessary, we should use fn **read_position** in stead